tfx/components/transform/executor.py

```diff
@@ -329,10 +329,20 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
         raise ValueError('analyze cannot be empty when splits_config is set.')
     else:
       splits_config.analyze.append('train')
-      splits_config.transform.extend(
-          artifact_utils.decode_split_names(
-              artifact_utils.get_single_instance(
-                  input_dict[EXAMPLES_KEY]).split_names))
+      split_names = artifact_utils.decode_split_names(
+          input_dict[EXAMPLES_KEY][0].split_names)
+
+      # Verify that all input artifacts have the same set of split names.
+      split_names_set = set(split_names)
+      for artifact in input_dict[EXAMPLES_KEY]:
```

Comment:
Could you make this a list comprehension so that we're iterating the artifact_split_names?